go/database/mpt/tool/export_test.go

```diff
@@ -0,0 +1,217 @@
+			t.Parallel()
+			dir := filepath.Join(t.TempDir(), name)
+			tf.createDB(t, dir)
+			ctx, cancel := context.WithCancel(context.Background())
+			catchInterrupt(ctx, cancel, time.Now())
+
+			success := make(chan any, 1)
+			writer := newMockWriter(t)
+
+			// first find number of writes
+			if err := tf.export(ctx, dir, writer); err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+
+			// save max count and reset number of writes
+			maxCount := writer.numOfWrites
+			writer.numOfWrites = 0
+
+			go func() {
+				if err := tf.export(ctx, dir, writer); err != nil {
+					t.Errorf("unexpected error: %v", err)
+					return
+				}
+				close(success)
+			}()
+
+			time.Sleep(200 * time.Millisecond)
+			err := syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+			if err != nil {
+				t.Error("failed to create a SIGINT signal")
+				return
+			}
+			select {
+			case <-success:
+				if maxCount > writer.numOfWrites {
+					break
+				}
+				t.Fatal("export was not interrupted")
+			case <-time.After(10 * time.Second):
+				t.Error("export was not interrupted")
+			}
+
+			err = tf.check(t, dir)
+			if err != nil {
+				t.Fatalf("db is corrupted after export interrupt: %v", err)
+			}
+		})
+	}
+
+}
+
+func checkExportedLiveDB(t *testing.T, dir string) error {
+	t.Helper()
+	return checkLiveDB(dir, cio.MptInfo{Config: mpt.S5LiveConfig})
+}
+func checkExportedArchive(t *testing.T, dir string) error {
+	t.Helper()
+	return checkLiveDB(dir, cio.MptInfo{Config: mpt.S5ArchiveConfig})
+}
+
+type mockWriter struct {
+	numOfWrites int
+}
+
+func newMockWriter(t *testing.T) *mockWriter {
+	t.Helper()
+	return &mockWriter{}
+}
+
+func (m *mockWriter) Write(_ []byte) (n int, err error) {
+	m.numOfWrites++
+	// slow down writting to be able to interrupt the export
+	time.Sleep(100 * time.Millisecond)
+	return 0, nil
+}
+
+func createLiveTrie(t *testing.T, dir string) {
+	t.Helper()
+	// Create a small LiveDB.
+	db, err := mpt.OpenGoFileState(dir, mpt.S5LiveConfig, 1024)
+	if err != nil {
+		t.Fatalf("failed to create test DB: %v", err)
+	}
+
+	addr1 := common.Address{1}
+	addr2 := common.Address{2}
+	key1 := common.Key{1}
+	key2 := common.Key{2}
+	value1 := common.Value{1}
+	value2 := common.Value{2}
+	err = errors.Join(
+		// First account, with code.
+		db.SetNonce(addr1, common.ToNonce(1)),
+		db.SetBalance(addr1, common.Balance{12}),
+		db.SetStorage(addr1, key1, value1),
+		db.SetCode(addr1, []byte("some_code")),
+		// Second account, without code.
+		db.SetNonce(addr2, common.ToNonce(2)),
+		db.SetBalance(addr2, common.Balance{14}),
+		db.SetStorage(addr2, key1, value1),
+		db.SetStorage(addr2, key2, value2),
+	)
+
+	if err != nil {
+		t.Fatalf("failed to seed test DB: %v", err)
+	}
+
+	_, err = db.GetHash()
+	if err != nil {
+		t.Fatalf("failed to fetch hash from DB: %v", err)
+	}
+	if err = db.Close(); err != nil {
+		t.Fatalf("failed to close DB: %v", err)
+	}
+}
+
+func createArchiveTrie(t *testing.T, dir string) {
```

Comment:
if the test is in the `io` package, there is already functionality to create test data